utility/src/main/java/org/oppia/android/util/locale/AndroidLocaleProfile.kt

```diff
@@ -100,17 +206,48 @@ data class AndroidLocaleProfile(val languageCode: String, val regionCode: String
       return maybeConstructProfile(languageCode, regionCode)
     }
 
+    /**
+     * Returns a new [AndroidLocaleProfile] using the provided [languageId]'s
+     * [LanguageId.getAndroidResourcesLanguageId] as the basis of the profile, or null if none can
+     * be created.
+     *
+     * This is meant to be used in cases when an [AndroidLocaleProfile] is needed to match a
+     * specific Android-compatible [Locale] (e.g. via [AndroidLocaleProfile.computeAndroidLocale])
+     * that can correctly match to specific Android app strings.
+     */
+    fun createFromAndroidResourcesLanguageId(languageId: LanguageId): AndroidLocaleProfile? {
+      val languageCode = languageId.androidResourcesLanguageId.languageCode
+      val regionCode = languageId.androidResourcesLanguageId.regionCode
+      return when {
+        !languageId.hasAndroidResourcesLanguageId() -> null
+        languageCode.isEmpty() -> null
+        // Empty region codes are allowed for Android resource IDs since they should always be used
+        // verbatim to ensure the correct Android resource string can be computed (such as for macro
+        // languages).
+        regionCode.isEmpty() -> LanguageAndWildcardRegionProfile(languageCode.asLowerCase())
+        else -> {
+          LanguageAndRegionProfile(
+            languageCode.asLowerCase(), regionCode.asLowerCase(), regionCode.asUpperCase()
+          )
+        }
+      }
+    }
+
     private fun maybeConstructProfile(
       languageCode: String,
       regionCode: String,
       emptyRegionAsWildcard: Boolean = false
     ): AndroidLocaleProfile? {
-      return if (languageCode.isNotEmpty() && (regionCode.isNotEmpty() || emptyRegionAsWildcard)) {
-        val adjustedRegionCode = if (emptyRegionAsWildcard && regionCode.isEmpty()) {
-          REGION_WILDCARD
-        } else regionCode
-        AndroidLocaleProfile(languageCode, adjustedRegionCode)
-      } else null
+      return when {
+        languageCode.isEmpty() -> null
```

Comment:
I don't think so since we have to be sure ``languageCode`` is set before the subsequent cases. The alternative would be doing ``&&`` checks for the next two cases with ``languageCode.isNotEmpty``, unless I'm missing something?

Reply:
I see why the language code needs to be non-empty.